server/protocols/webadmin/webadmin-mailqueue/src/main/java/org/apache/james/webadmin/service/DeleteMailsFromMailQueueTask.java

```diff
@@ -80,7 +86,82 @@ public long getInitialCount() {
         }
     }
 
+    public static class UnknownSerializedQueue extends RuntimeException {
+        public UnknownSerializedQueue(String queueName) {
+            super("Unable to retrieve '" + queueName + "' queue");
+        }
+    }
+
+    private static class DeleteMailsFromFailQueueTaskDTO implements TaskDTO {
+
+        private final String type;
+        private final String queue;
+        private final String sender;
+        private final String name;
+        private final String recipient;
+
+        public DeleteMailsFromFailQueueTaskDTO(@JsonProperty("type") String type,
+                                               @JsonProperty("queue") String queue,
+                                               @JsonProperty("sender") String sender,
+                                               @JsonProperty("name") String name,
+                                               @JsonProperty("recipient") String recipient) {
+            this.type = type;
+            this.queue = queue;
+            this.sender = sender;
+            this.name = name;
+            this.recipient = recipient;
+        }
+
+        public static DeleteMailsFromFailQueueTaskDTO toDTO(DeleteMailsFromMailQueueTask domainObject, String typeName) {
+            return new DeleteMailsFromFailQueueTaskDTO(
+                typeName,
+                domainObject.queue.getName(),
+                domainObject.maybeSender.map(MailAddress::asString).orElse(null),
```

Comment:
why did you chose null instead of Optional?

Reply:
I was not sure that jackson was dealing with `Optional` correctly, does it?

Reply:
it does if you add the java 8 jackson module